fix: Refactor communication date copy logic for DB compatibility

abhishek0010 opened the pull request:
Updated the execute function to handle PostgreSQL and MySQL/MariaDB syntax for copying communication dates.

The current version 15 is throwing errors upon migrate commands
<img width="1727" height="945" alt="Screenshot 2025-12-16 at 16 21 17" src="https://github.com/user-attachments/assets/7573911b-4ea7-47d2-9ae6-b945e01ee8a3" />

I request you to add this to the new version of frappe as soon as possible.


Closes https://github.com/frappe/frappe/issues/35281

abhishek0010 committed:
fix: Refactor communication date copy logic for DB compatibility

Updated the execute function to handle PostgreSQL and MySQL/MariaDB syntax for copying communication dates.

abhishek0010 committed:
Refactor communication date copy logic for DB compatibility

Updated the execute function to handle PostgreSQL and MySQL/MariaDB syntax for copying communication dates.

abhishek0010 commented:
@NagariaHussain  This issue was with this [PR](https://github.com/frappe/frappe/pull/35235)

akhilnarang reviewed (commented):
Can use `frappe.db.multisql` here, or try to rewrite the query using query builder.

abhishek0010 committed:
fix: updated file to use multisql

abhishek0010 commented:
@akhilnarang I have tried using `frappe.db.multisql`

NagariaHussain commented:
As Ankush suggested, let's use query builder.

abhishek0010 commented:
Does query builder support `UPDATE` statements for both postgres and mariadb? @akhilnarang @NagariaHussain
I have already updated this pull request to use `frappe.db.multisql`

NagariaHussain commented:
You just write the code once, Query Builder will output the right SQL depending on the database.

abhishek0010 commented:
@akhilnarang @NagariaHussain I have updated the patch file to use `query_builder`.

abhishek0010 commented:
I saw that https://github.com/frappe/frappe/pull/35224/ is also there resolving the same issue. Shall I close this PR?

abhishek0010 committed:
fix: fixed whitespace issue for github actions

abhishek0010 commented:
I have updated the commit names and the patch file to satisfy the github action checks. @akhilnarang @NagariaHussain
There is one more issue here. Now the patch is failing some sqlite cases when running Tests / Success action

```

		╔══════════════════════════════════��═════════╗
		║   Parallel Test Runner Execution Summary   ║
		╠════════════════════════════════════════════╣
		║ Mode:           Parallel                   ║
		║ App:            frappe                     ║
		║ Site:           test_site                  ║
		║ Build Number:   2                          ║
		║ Total Builds:   2                          ║
			║ Coverage Rep.:  {cc.outfile:<26} ║
		╚════════════════════════════════════════════╝
		
.....

frappe.tests.test_boilerplate.TestBoilerPlate
    ✔  test_create_app
    ✔  test_create_app_without_git_init
Building Index.....Building Index.....Building Index.....Building Index.....Building Index.....Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 104, in application
    validate_auth()
    ~~~~~~~~~~~~~^^
  File "apps/frappe/frappe/auth.py", line 635, in validate_auth
    validate_auth_via_api_keys(authorization_header)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/auth.py", line 701, in validate_auth_via_api_keys
    validate_api_key_secret(api_key, api_secret, authorization_source)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/auth.py", line 733, in validate_api_key_secret
    raise frappe.AuthenticationError
frappe.exceptions.AuthenticationError

'test_app' created at /home/runner/work/frappe/frappe/apps/test_app
'test_app_no_git' created at /home/runner/work/frappe/frappe/apps/test_app_no_git
App Title [Test App]: App Title should start with a letter and it can only consist of letters, numbers, spaces and underscores
    ✔  test_invalid_inputs
App Title [Test App]: App Description: App Publisher: App Email: App Email should be a valid email address.
App Email: App Email should be a valid email address.
App Email: App License (agpl-3.0, apache-2.0, bsd-2-clause, bsd-3-clause, bsl-1.0, cc0-1.0, epl-2.0, gpl-2.0, gpl-3.0, lgpl-2.1, mit, mpl-2.0, unlicense) [mit]: Create GitHub Workflow action for unittests [y/N]: Branch Name [HEAD]: Select app for new patch (frappe): Provide DocType name on which this patch will apply: Describe what this patch does: Provide filename for this patch [delete_all_users.py]: Patch folder '/home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/user/patches' doesn't exist, create it? [Y/n]: Created /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/user/patches/delete_all_users.py and updated patches.txt
    ✔  test_new_patch_util
    ✔  test_simple_input_to_boilerplate
    ✔  test_valid_ci_yaml

frappe.tests.test_boot.TestBootData
    ✔  test_get_unseen_notes

frappe.tests.test_boot.TestPermissionQueries
    ✔  test_get_user_pages_or_reports_with_permission_query

frappe.tests.test_caching.TestCachingUtils
    ✔  test_request_cache

frappe.tests.test_caching.TestDocumentCache
    ✔  test_cache_invalidation_set_value
    ✔  test_caching

frappe.tests.test_caching.TestHttpCache
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/threading.py", line 1044, in _bootstrap
    self._bootstrap_inner()
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/threading.py", line 1082, in _bootstrap_inner
    self._context.run(self.run)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/tests/test_api.py", line 72, in run
    self._return = self._target(*self._args, **self._kwargs)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/werkzeug/test.py", line 1162, in get
    return self.open(*args, **kw)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/werkzeug/test.py", line 1116, in open
    response_parts = self.run_wsgi_app(request.environ, buffered=buffered)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/werkzeug/test.py", line 988, in run_wsgi_app
    rv = run_wsgi_app(self.application, environ, buffered=buffered)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/werkzeug/test.py", line 1264, in run_wsgi_app
    app_rv = app(environ, start_response)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/app.py", line 84, in application
    app(environ, start_response),
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/werkzeug/wrappers/request.py", line 193, in application
    resp = f(*args[:-2] + (request,))
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/app.py", line 121, in application
    response = frappe.api.handle(request)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/api/__init__.py", line 63, in handle
    data = endpoint(**arguments)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/api/v1.py", line 40, in handle_rpc_call
    return frappe.handler.handle()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/handler.py", line 53, in handle
    data = execute_cmd(cmd)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/handler.py", line 86, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/__init__.py", line 1126, in call
    return fn(*args, **newargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/typing_validations.py", line 36, in wrapper
    return func(*args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/caching.py", line 248, in inner
    ret = func(*args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/client.py", line 396, in is_document_amended
    return frappe.db.exists(doctype, {"amended_from": docname})
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/database.py", line 1285, in exists
    return self.get_value(dt, dn, ignore=True, cache=cache, order_by=None, debug=debug)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/database.py", line 555, in get_value
    result = self.get_values(
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/database.py", line 667, in get_values
    out = query.run(as_dict=as_dict, debug=debug, update=update, run=run, pluck=pluck)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/query_builder/utils.py", line 131, in execute_query
    result = frappe.local.db.sql(query, params, *args, **kwargs)  # nosemgrep
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/postgres/database.py", line 233, in sql
    return super().sql(modify_query(query), modify_values(values), *args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/database.py", line 294, in sql
    traceback.print_stack()
Error in query:
column "amended_from" does not exist
LINE 1: SELECT "name" FROM "tabUser" WHERE "amended_from"='Guest' LI...
                                           ^

  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/threading.py", line 1044, in _bootstrap
    self._bootstrap_inner()
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/threading.py", line 1082, in _bootstrap_inner
    self._context.run(self.run)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/tests/test_api.py", line 72, in run
    self._return = self._target(*self._args, **self._kwargs)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/werkzeug/test.py", line 1162, in get
    return self.open(*args, **kw)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/werkzeug/test.py", line 1116, in open
    response_parts = self.run_wsgi_app(request.environ, buffered=buffered)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/werkzeug/test.py", line 988, in run_wsgi_app
    rv = run_wsgi_app(self.application, environ, buffered=buffered)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/werkzeug/test.py", line 1264, in run_wsgi_app
    app_rv = app(environ, start_response)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/app.py", line 84, in application
    app(environ, start_response),
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/werkzeug/wrappers/request.py", line 193, in application
    resp = f(*args[:-2] + (request,))
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/app.py", line 121, in application
    response = frappe.api.handle(request)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/api/__init__.py", line 77, in handle
    app_name = get_doctype_app_map().get(doctype)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/caching.py", line 157, in site_cache_wrapper
    result = func(*args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/modules/utils.py", line 293, in get_doctype_app_map
    .run()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/query_builder/utils.py", line 131, in execute_query
    result = frappe.local.db.sql(query, params, *args, **kwargs)  # nosemgrep
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/postgres/database.py", line 233, in sql
    return super().sql(modify_query(query), modify_values(values), *args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/database.py", line 294, in sql
    traceback.print_stack()
Error in query:
current transaction is aborted, commands ignored until end of transaction block

    ✔  test_http_headers

.......

frappe.tests.test_db.TestDDLCommandsMaria
    =  test_add_index
    =  test_change_type
    =  test_describe
    =  test_rename

frappe.tests.test_db.TestDDLCommandsPost
    ✔  test_add_index
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/bench_helper.py", line 177, in <module>
    main()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/bench_helper.py", line 77, in main
    FrappeClickWrapper(click.Group, handle_exception)(commands=commands)(prog_name="bench")
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1442, in __call__
    return self.main(*args, **kwargs)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1363, in main
    rv = self.invoke(ctx)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/bench_helper.py", line 48, in invoke
    return super().invoke(ctx)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/bench_helper.py", line 48, in invoke
    return super().invoke(ctx)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1226, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 794, in invoke
    return callback(*args, **kwargs)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/decorators.py", line 34, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/commands/__init__.py", line 28, in _func
    ret = f(ctx.obj, *args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/commands/testing.py", line 425, in run_parallel_tests
    runner.setup_and_run()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/parallel_test_runner.py", line 46, in setup_and_run
    self.run_tests()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/parallel_test_runner.py", line 87, in run_tests
    self.run_tests_for_file(test_file_info)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/parallel_test_runner.py", line 118, in run_tests_for_file
    test_suite(self.test_result)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 122, in run
    test(result)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 122, in run
    test(result)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 122, in run
    test(result)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/case.py", line 725, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/case.py", line 669, in run
    self._callTestMethod(testMethod)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/case.py", line 615, in _callTestMethod
    result = method()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/tests/test_db.py", line 1008, in test_change_type
    frappe.db.change_column_type(self.test_table_name, "id", "bigint")
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/postgres/database.py", line 312, in change_column_type
    return self.sql_ddl(
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/database.py", line 451, in sql_ddl
    self.sql(query, debug=debug)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/postgres/database.py", line 233, in sql
    return super().sql(modify_query(query), modify_values(values), *args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/database.py", line 294, in sql
    traceback.print_stack()
Error in query:
column "id" cannot be cast automatically to type bigint
HINT:  You might need to specify "USING id::bigint".

    ✔  test_change_type
    ✔  test_describe
    ✔  test_is
    ✔  test_rename
    ✔  test_sequence_table_creation

frappe.tests.test_db.TestDbConnectWithEnvCredentials
    ✔  test_connect_fails_with_wrong_credentials_by_env

frappe.tests.test_db.TestPostgresSchemaQueryIndependence
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/bench_helper.py", line 177, in <module>
    main()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/bench_helper.py", line 77, in main
    FrappeClickWrapper(click.Group, handle_exception)(commands=commands)(prog_name="bench")
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1442, in __call__
    return self.main(*args, **kwargs)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1363, in main
    rv = self.invoke(ctx)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/bench_helper.py", line 48, in invoke
    return super().invoke(ctx)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/bench_helper.py", line 48, in invoke
    return super().invoke(ctx)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1226, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 794, in invoke
    return callback(*args, **kwargs)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/decorators.py", line 34, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/commands/__init__.py", line 28, in _func
    ret = f(ctx.obj, *args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/commands/testing.py", line 425, in run_parallel_tests
    runner.setup_and_run()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/parallel_test_runner.py", line 46, in setup_and_run
    self.run_tests()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/parallel_test_runner.py", line 87, in run_tests
    self.run_tests_for_file(test_file_info)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/parallel_test_runner.py", line 118, in run_tests_for_file
    test_suite(self.test_result)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 122, in run
    test(result)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 122, in run
    test(result)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 122, in run
    test(result)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/case.py", line 725, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/case.py", line 669, in run
    self._callTestMethod(testMethod)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/case.py", line 615, in _callTestMethod
    result = method()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/tests/test_db.py", line 1339, in test_add_index
    frappe.db.add_index(doctype="User", fields=("col_c",))
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/postgres/database.py", line 401, in add_index
    self.sql_ddl(
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/database.py", line 451, in sql_ddl
    self.sql(query, debug=debug)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/postgres/database.py", line 233, in sql
    return super().sql(modify_query(query), modify_values(values), *args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/database.py", line 294, in sql
    traceback.print_stack()
Error in query:
column "col_c" does not exist

    ✔  test_add_index
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/bench_helper.py", line 177, in <module>
    main()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/bench_helper.py", line 77, in main
    FrappeClickWrapper(click.Group, handle_exception)(commands=commands)(prog_name="bench")
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1442, in __call__
    return self.main(*args, **kwargs)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1363, in main
    rv = self.invoke(ctx)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/bench_helper.py", line 48, in invoke
    return super().invoke(ctx)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/bench_helper.py", line 48, in invoke
    return super().invoke(ctx)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1226, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 794, in invoke
    return callback(*args, **kwargs)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/decorators.py", line 34, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/commands/__init__.py", line 28, in _func
    ret = f(ctx.obj, *args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/commands/testing.py", line 425, in run_parallel_tests
    runner.setup_and_run()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/parallel_test_runner.py", line 46, in setup_and_run
    self.run_tests()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/parallel_test_runner.py", line 87, in run_tests
    self.run_tests_for_file(test_file_info)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/parallel_test_runner.py", line 118, in run_tests_for_file
    test_suite(self.test_result)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 122, in run
    test(result)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 122, in run
    test(result)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 122, in run
    test(result)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/case.py", line 725, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/case.py", line 669, in run
    self._callTestMethod(testMethod)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/case.py", line 615, in _callTestMethod
    result = method()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/tests/test_db.py", line 1348, in test_add_unique
    frappe.db.add_unique(f"{self.test_table_name}", ["col_c", "col_d"])
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/postgres/database.py", line 423, in add_unique
    self.sql(
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/postgres/database.py", line 233, in sql
    return super().sql(modify_query(query), modify_values(values), *args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/database.py", line 294, in sql
    traceback.print_stack()
Error in query:
column "col_c" named in key does not exist

    ✔  test_add_unique
    ✔  test_db_table_columns
    ✔  test_describe
    ✔  test_get_column_type
    ✔  test_get_table_columns_description
    ✔  test_get_tables
    ✔  test_has_index
    ✔  test_search_path



.........

frappe.tests.test_redis.TestRedisAuth
    ✔  test_adding_redis_user
    ✔  test_rq_gen_acllist
    ✔  test_rq_namespace

frappe.tests.test_removing_orphans.TestRemovingOrphans
App Title [Test App]: App Description: App Publisher: App Email: App License (agpl-3.0, apache-2.0, bsd-2-clause, bsd-3-clause, bsl-1.0, cc0-1.0, epl-2.0, gpl-2.0, gpl-3.0, lgpl-2.1, mit, mpl-2.0, unlicense) [mit]: Create GitHub Workflow action for unittests [y/N]: Branch Name [HEAD]: Wrote document file for DocType Test Child Table With Link To Doctype A at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_a/test_child_table_with_link_to_doctype_a.json
Wrote document file for DocType Test Child Table With Link To Doctype B at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_b/test_child_table_with_link_to_doctype_b.json
Wrote document file for DocType Test Doctype A With Child Table With Link To Doctype B at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_doctype_a_with_child_table_with_link_to_doctype_b/test_doctype_a_with_child_table_with_link_to_doctype_b.json
Wrote document file for DocType Test Doctype B With Child Table With Link To Doctype A at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_doctype_b_with_child_table_with_link_to_doctype_a/test_doctype_b_with_child_table_with_link_to_doctype_a.json
Wrote document file for DocType Test Doctype D at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_doctype_d/test_doctype_d.json
Wrote document file for DocType Test Doctype E at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_doctype_e/test_doctype_e.json
Wrote document file for DocType Test Child Table With Link To Doctype A at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_a/test_child_table_with_link_to_doctype_a.json
/home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_a/test_child_table_with_link_to_doctype_a.py already exists, skipping...
Wrote document file for DocType Test Child Table With Link To Doctype B at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_b/test_child_table_with_link_to_doctype_b.json
/home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_b/test_child_table_with_link_to_doctype_b.py already exists, skipping...
Wrote document file for DocType Test Child Table With Link To Doctype A at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_a/test_child_table_with_link_to_doctype_a.json
Wrote document file for DocType Test Child Table With Link To Doctype B at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_b/test_child_table_with_link_to_doctype_b.json
Wrote document file for DocType Test Doctype A With Child Table With Link To Doctype B at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_doctype_a_with_child_table_with_link_to_doctype_b/test_doctype_a_with_child_table_with_link_to_doctype_b.json
Wrote document file for DocType Test Doctype B With Child Table With Link To Doctype A at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_doctype_b_with_child_table_with_link_to_doctype_a/test_doctype_b_with_child_table_with_link_to_doctype_a.json
Wrote document file for DocType Test Doctype D at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_doctype_d/test_doctype_d.json
Wrote document file for DocType Test Doctype E at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_doctype_e/test_doctype_e.json
Wrote document file for DocType Test Child Table With Link To Doctype A at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_a/test_child_table_with_link_to_doctype_a.json
/home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_a/test_child_table_with_link_to_doctype_a.py already exists, skipping...
Wrote document file for DocType Test Child Table With Link To Doctype B at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_b/test_child_table_with_link_to_doctype_b.json
    ✔  test_removing_orphan
/home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_b/test_child_table_with_link_to_doctype_b.py already exists, skipping...
Wrote document file for DocType Test Child Table With Link To Doctype A at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_a/test_child_table_with_link_to_doctype_a.json
Wrote document file for DocType Test Child Table With Link To Doctype B at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_b/test_child_table_with_link_to_doctype_b.json
Wrote document file for DocType Test Doctype A With Child Table With Link To Doctype B at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_doctype_a_with_child_table_with_link_to_doctype_b/test_doctype_a_with_child_table_with_link_to_doctype_b.json
Wrote document file for DocType Test Doctype B With Child Table With Link To Doctype A at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_doctype_b_with_child_table_with_link_to_doctype_a/test_doctype_b_with_child_table_with_link_to_doctype_a.json
Wrote document file for DocType Test Doctype D at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_doctype_d/test_doctype_d.json
Wrote document file for DocType Test Doctype E at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_doctype_e/test_doctype_e.json
Wrote document file for DocType Test Child Table With Link To Doctype A at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_a/test_child_table_with_link_to_doctype_a.json
/home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_a/test_child_table_with_link_to_doctype_a.py already exists, skipping...
Wrote document file for DocType Test Child Table With Link To Doctype B at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_b/test_child_table_with_link_to_doctype_b.json
/home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_child_table_with_link_to_doctype_b/test_child_table_with_link_to_doctype_b.py already exists, skipping...
Wrote document file for DocType Test WebViewDocType at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_webviewdoctype/test_webviewdoctype.json
Wrote document file for DocType Test WebViewDocType at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_webviewdoctype/test_webviewdoctype.json
/home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_webviewdoctype/test_webviewdoctype.py already exists, skipping...
/home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_webviewdoctype/test_test_webviewdoctype.py already exists, skipping...
/home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_webviewdoctype/test_webviewdoctype.js already exists, skipping...
Wrote document file for DocType Test WebViewDocType at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_webviewdoctype/test_webviewdoctype.json
/home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_webviewdoctype/test_webviewdoctype.py already exists, skipping...
/home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_webviewdoctype/test_test_webviewdoctype.py already exists, skipping...
/home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_webviewdoctype/test_webviewdoctype.js already exists, skipping...
Update global search for all web pages...
Exporting User app frappe filters None
Exporting Contact app frappe filters None
Exporting Role app frappe filters None
Exporting User app frappe filters None
Exporting Contact app frappe filters None
Exporting Role app frappe filters None
Exporting User app frappe filters None
Exporting Contact app frappe filters None
Exporting Role app frappe filters None
Exporting User app frappe filters None
Exporting Contact app frappe filters None
Exporting Role app frappe filters None
Exporting Document Note ai0hlho16o
Wrote document file for Note ai0hlho16o at /home/runner/work/frappe/frappe/apps/frappe/frappe/desk/note/ai0hlho16o/ai0hlho16o.json
Wrote document file for DocType Test WebViewDocType at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_webviewdoctype/test_webviewdoctype.json
Wrote document file for DocType Test uwxdskafpq at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/test_uwxdskafpq/test_uwxdskafpq.json
Updating customizations for Note
Recorded queries, open recorder to view them.
Wrote document file for Report Orphan Report at /home/runner/work/frappe/frappe/apps/frappe/frappe/custom/report/orphan_report/orphan_report.json
Created report: Orphan Report
Removing orphan Workspaces
Removing orphan Dashboards
Removing orphan Pages
Removing orphan Reports
Deleting entity Report Orphan Report

......

frappe.tests.test_seen.TestSeen
    ✔  test_if_user_is_added

frappe.tests.test_sequence.TestSequence
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/bench_helper.py", line 177, in <module>
    main()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/bench_helper.py", line 77, in main
    FrappeClickWrapper(click.Group, handle_exception)(commands=commands)(prog_name="bench")
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1442, in __call__
    return self.main(*args, **kwargs)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1363, in main
    rv = self.invoke(ctx)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/bench_helper.py", line 48, in invoke
    return super().invoke(ctx)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/utils/bench_helper.py", line 48, in invoke
    return super().invoke(ctx)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 1226, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/core.py", line 794, in invoke
    return callback(*args, **kwargs)
  File "/home/runner/work/frappe/frappe/env/lib/python3.14/site-packages/click/decorators.py", line 34, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/commands/__init__.py", line 28, in _func
    ret = f(ctx.obj, *args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/commands/testing.py", line 425, in run_parallel_tests
    runner.setup_and_run()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/parallel_test_runner.py", line 46, in setup_and_run
    self.run_tests()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/parallel_test_runner.py", line 87, in run_tests
    self.run_tests_for_file(test_file_info)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/parallel_test_runner.py", line 118, in run_tests_for_file
    test_suite(self.test_result)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 122, in run
    test(result)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 122, in run
    test(result)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/suite.py", line 122, in run
    test(result)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/case.py", line 725, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/case.py", line 669, in run
    self._callTestMethod(testMethod)
  File "/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/unittest/case.py", line 615, in _callTestMethod
    result = method()
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/tests/test_sequence.py", line 34, in test_create_sequence
    frappe.db.get_next_sequence_val(seq_name)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/database.py", line 1524, in get_next_sequence_val
    return get_next_val(*args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/sequence.py", line 84, in get_next_val
    return db.sql(f"SELECT nextval({sequence_name})")[0][0]
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/postgres/database.py", line 233, in sql
    return super().sql(modify_query(query), modify_values(values), *args, **kwargs)
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/database/database.py", line 294, in sql
    traceback.print_stack()
Error in query:
nextval: reached maximum value of sequence "test_create_sequence_5dea7_id_seq" (2)

    ✔  test_create_sequence
    ✔  test_set_next_val

frappe.tests.test_sitemap.TestSitemap
    ✔  test_sitemap

frappe.tests.test_sqlite_search.TestSQLiteSearchAPI
    ✖  test_advanced_scoring_and_ranking
    ✔  test_background_operations
    ✔  test_basic_search_functionality
    ✔  test_configuration_and_schema_validation
    ✔  test_content_processing_and_html_handling
    ✔  test_document_indexing_operations
    ✔  test_index_lifecycle_and_status_methods
    ✔  test_search_disabled_state
    ✔  test_search_filtering_and_permissions
    ✔  test_search_result_summary_and_metadata
    ✔  test_spelling_correction_and_query_expansion

........

frappe.website.doctype.website_theme.test_website_theme.TestWebsiteTheme
    ✔  test_after_migrate_hook  (8.55s)
    ✔  test_backup_files  (9.34s)
    ✔  test_get_scss_paths
    ✔  test_imports_to_ignore
    ✔  test_website_theme  (2.35s)

frappe.workflow.doctype.workflow.test_workflow.TestWorkflow
    ✔  test_approve
    ✔  test_default_condition
    ✔  test_get_common_transition_actions
    ✔  test_if_workflow_actions_were_processed_using_role
    ✔  test_if_workflow_set_on_action
    ✔  test_sync_tasks
    ✔  test_syntax_error_in_transition_rule
    ✔  test_workflow_condition
    ✔  test_wrong_action
Deleting orphaned Report.
Wrote document file for DocType Test Rename Document Old at /home/runner/work/frappe/frappe/apps/frappe/frappe/custom/doctype/test_rename_document_old/test_rename_document_old.json
Enqueuing frappe.tests.test_sqlite_search.TestSQLiteSearch.build_index
Sleeping for 2 secs to confirm token expires..
Wrote document file for DocType VirtualDoctypeTestChild at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/virtualdoctypetestchild/virtualdoctypetestchild.json
Wrote document file for DocType VirtualDoctypeTest at /home/runner/work/frappe/frappe/apps/frappe/frappe/core/doctype/virtualdoctypetest/virtualdoctypetest.json


======================================================================
 FAIL  test_advanced_scoring_and_ranking (frappe.tests.test_sqlite_search.TestSQLiteSearchAPI.test_advanced_scoring_and_ranking)
Test scoring pipeline, ranking, and result ordering.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/frappe/frappe/apps/frappe/frappe/tests/test_sqlite_search.py", line 278, in test_advanced_scoring_and_ranking
    self.assertGreater(result["score"], 1.0)
    ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
    i = 2
    result = {'id': 'Note:ebk2o6fngg', 'score': 0.9884410848696512, 'original_rank': 1, 'bm25_score': -4.058470430394307, 'title': 'Python <mark>Programming</mark> Guide', 'content': 'Learn Python basics and advanced concepts', 'doctype': 'Note', 'name': 'ebk2o6fngg', 'author': 'Administrator', 'modified': 1766368445.08747, 'modified_rank': 1}
    results = {'results': [{'id': 'Note:ebk2o6fngg', 'score': 0.9884410848696512, 'original_rank': 1, 'bm25_score': -4.058470430394307, 'title': 'Python <mark>Programming</mark> Guide', 'content': 'Learn Python basics and advanced concepts', 'doctype': 'Note', 'name': 'ebk2o6fngg', 'author': 'Administrator', 'modified': 1766368445.08747, 'modified_rank': 1}], 'summary': {'duration': 0.001, 'total_matches': 1, 'returned_matches': 1, 'corrected_words': {'Programming': 'programming'}, 'corrected_query': 'programming', 'title_only': False, 'filtered_matches': 1, 'applied_filters': {}}}
    scores = [1.0322766768625597, 0.9709313287983129, 0.25108113190411746]
    self = <frappe.tests.test_sqlite_search.TestSQLiteSearchAPI testMethod=test_advanced_scoring_and_ranking>
    title_match_found = True
AssertionError: 0.9884410848696512 not greater than 1.0

Tests: 1001, Failing: 1, Errors: 0
Saved Coverage
Error: Process completed with exit code 1.
```

abhishek0010 commented:
Can you fast track this update? I am running v15.90.0 since past month due to this. Thanks in advance

ankush commented on frappe/core/doctype/communication_link/patches/copy_communication_date_to_link.py:
```diff
@@ -1,32 +1,35 @@
 			break
+
+		# Update records in batch
+		for record in records:
```
This is a bad idea? Communication have a huge amount of rows, updating them one at a time isn't good.

POstgres does support bulk update.. why are doing them like this?

HichamDz38 replied:
@ankush a pull request for that issue already opened two days before this pr creation
and it follow all your review notes
**[fix: support postgres database for the communication link patch #35224](https://github.com/frappe/frappe/pull/35224)**

abhishek0010 replied:
@ankush 
First of all I am not the creator of this patch. I tried to update https://github.com/frappe/frappe/pull/35235 created by @NagariaHussain which broke my `bench migrate` command for my custom app running on latest frappe.


Second: Does frappe `query builder` support bulk update operation? I used `multisql` before but @NagariaHussain and @akhilnarang instructed to use frappe query builder. I tried to accomodate the implemented functions within `frappe.qb` to be used in this updated patch file. I can revert back the pacth file to use `multisql`:

```
import frappe

  def execute():
      batch_size = 10_000

      while True:
          # Update communication_date in batches
          frappe.db.multisql(
              {
                  "postgres": """
                      UPDATE "tabCommunication Link" cl
                      SET communication_date = c.communication_date
                      FROM "tabCommunication" c
                      WHERE cl.parent = c.name
                      AND cl.communication_date IS NULL
                      AND c.communication_date IS NOT NULL
                      LIMIT %s
                  """,
                  "mariadb": """
                      UPDATE `tabCommunication Link` cl
                      INNER JOIN `tabCommunication` c ON cl.parent = c.name
                      SET cl.communication_date = c.communication_date
                      WHERE cl.communication_date IS NULL
                      AND c.communication_date IS NOT NULL
                      LIMIT %s
                  """,
              },
              values=(batch_size,),
          )

          frappe.db.commit()

          # Check if more rows need updating
          check = frappe.db.multisql(
              {
                  "postgres": """
                      SELECT 1 FROM "tabCommunication Link" cl
                      INNER JOIN "tabCommunication" c ON cl.parent = c.name
                      WHERE cl.communication_date IS NULL
                      AND c.communication_date IS NOT NULL
                      LIMIT 1
                  """,
                  "mariadb": """
                      SELECT 1 FROM `tabCommunication Link` cl
                      INNER JOIN `tabCommunication` c ON cl.parent = c.name
                      WHERE cl.communication_date IS NULL
                      AND c.communication_date IS NOT NULL
                      LIMIT 1
                  """,
              }
          )

          if not check:
              break
```

Shall I create a new commit with `multisql`? @ankush

ankush commented:
@AarDG10 can you take this forward... PG related

abhishek0010 commented:
Shall I close this PR? @ankush 

@AarDG10 has already made progress for this patch here https://github.com/frappe/frappe/pull/35224

ankush commented:
https://github.com/frappe/frappe/pull/35488
